File: scratch-runtime/src/main/java/io/github/gaming32/scratch2jvm/runtime/ScratchCostume.java
package io.github.gaming32.scratch2jvm.runtime;

public final class ScratchCostume {
    @SuppressWarnings("unused")
    public enum Format {
        SVG, PNG
    }

    public final String name;
    public final String path;
    public final Format format;
    public final double centerX, centerY;
    public final double coordinateScale;
    public double width, height;

    public ScratchCostume(
        String name,
        String path,
        Format format,
        double centerX,
        double centerY,
        double coordinateScale
    ) {
        this.name = name;
        this.path = path;
        this.format = format;
        this.centerX = centerX;
        this.centerY = centerY;
        this.coordinateScale = coordinateScale;
    }
}

File: scratch-parser/src/main/kotlin/io/github/gaming32/scratch2jvm/parser/data/ScratchCostume.kt
package io.github.gaming32.scratch2jvm.parser.data

import com.google.gson.JsonObject
import io.github.gaming32.scratch2jvm.parser.PrettyPrintable

public data class ScratchCostume(
    public val name: String,
    public val path: String,
    public val format: Format,
    public val centerX: Double,
    public val centerY: Double
) : PrettyPrintable {
    public enum class Format {
        SVG, PNG
    }

    public companion object {
        public fun fromJson(data: JsonObject): ScratchCostume = ScratchCostume(
            name = data["name"].asString,
            path = "/" + data["md5ext"].asString,
            format = Format.valueOf(data["dataFormat"].asString.uppercase()),
            centerX = data["rotationCenterX"].asDouble,
            centerY = data["rotationCenterY"].asDouble
        )
    }
}
